lib/ex_unit/lib/ex_unit/capture_log.ex

```diff
@@ -84,11 +84,11 @@ defmodule ExUnit.CaptureLog do
   end
 
   defp add_capture(pid, opts) do
-    GenEvent.add_mon_handler(Logger, {Console, pid}, {pid, opts})
+    :gen_event.add_sup_handler(Logger, {Console, pid}, {pid, opts})
```

Comment:
Could there be a side effect of linking here? I think will we need to unlink from `Logger` after `delete_handler` as `:gen_event` doesnt remove the link.